homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -36,41 +35,20 @@
 ICON = "mdi:gas-station"
 
 
-async def async_setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the tankerkoenig sensors."""
 
-    if discovery_info is None:
-        return
-
-    tankerkoenig = hass.data[DOMAIN]
-
-    async def async_update_data():
-        """Fetch data from API endpoint."""
-        try:
-            return await tankerkoenig.fetch_data()
-        except LookupError as err:
-            raise UpdateFailed("Failed to fetch data") from err
-
-    coordinator = DataUpdateCoordinator(
-        hass,
-        _LOGGER,
-        name=NAME,
-        update_method=async_update_data,
-        update_interval=tankerkoenig.update_interval,
-    )
+    coordinator: TankerkoenigDataUpdateCoordinator = hass.data[DOMAIN][entry.unique_id]
 
     # Fetch initial data so we have data when entities subscribe
     await coordinator.async_refresh()
```

Comment:
Is this needed? You call `coordinator.async_config_entry_first_refresh` in `async_setup_entry`

Reply:
ahh ... no this is obsolet, since the coordinator setup was moved to `async_setup_entry` ... will create a PR for that